solidity-v1/dashboard/src/pages/threshold/ThresholdUpgradePage.jsx

```diff
@@ -0,0 +1,316 @@
+        <AllocationProgressBar
+          title={"available grant allocation"}
+          currentValue={totalGrantedUnstakedBalance}
+          totalValue={notStakedTotalAmount}
+          className={"mb-2"}
+          isDataFetching={isDataFetching}
+        />
+        <div className="upgrade-not-staked">
+          <h4 className={"mb-1"}>Upgrade Not Staked Tokens</h4>
+          <UpgradeTokensTile
+            title={"Wallet"}
+            btnText={"upgrade to t"}
+            className={"mb-1"}
+            isLink
+            buttonDisabled={isDataFetching}
+            renderButton={() => (
+              <UpgradeTokensTile.Link
+                to={"https://google.com"}
+                linkText={"upgrade to t"}
+              />
+            )}
+          >
+            <UpgradeTokensTile.Row
+              label={"Liquid KEEP"}
+              amount={totalOwnedUnstakedBalance}
+              isDataFetching={isDataFetching}
+            />
+          </UpgradeTokensTile>
+          <UpgradeTokensTile
+            title={"Grant Allocation"}
+            titleTooltipProps={
+              resourceTooltipProps.thresholdPageGrantAllocation
+            }
+            renderButton={() => (
+              <UpgradeTokensTile.Button
+                btnText={"withdraw from grant"}
+                buttonDisabled={
+                  lte(totalGrantedUnstakedBalance, 0) || isDataFetching
+                }
+                onBtnClick={onWithdrawFromGrant}
+              />
+            )}
+          >
+            <UpgradeTokensTile.Row
+              label={"Available KEEP"}
+              amount={totalGrantedUnstakedBalance}
+              isDataFetching={isDataFetching}
+            />
+          </UpgradeTokensTile>
+        </div>
+      </section>
+
+      <section className="tile staked">
+        <div className="staked__title-container">
+          <h3 className="staked__title">Staked</h3>
+          <div className="staked__additional-info">
+            <span className="staked__additional-info-row mr-2">
+              <Icons.Success
+                width={16}
+                height={16}
+                className="staked__additional-info-icon staked__additional-info-icon--color-green"
+              />{" "}
+              ECDSA
+            </span>
+            <span className="staked__additional-info-row">
+              <Icons.Success
+                width={16}
+                height={16}
+                className="staked__additional-info-icon staked__additional-info-icon--color-green"
+              />{" "}
+              Random Beacon
+            </span>
+          </div>
+        </div>
+        <OnlyIf condition={isDataFetching}>
+          <div className="staked__token-amount staked__token-amount--loading">
+            <Icons.KeepOutline
+              width={32}
+              height={32}
+              className={`token-amount__icon`}
+            />
+            <Skeleton tag="h2" shining color="grey-20" width="40%" />
```

Comment:
Maybe we can use `TokenAmountSkeleton` component?

Reply:
Fixed in e3d3a0a44fba2c556ed2ffa14ed7d5b5819534c6